### ISSUE TITLE
Favorability Check Kinderfreibetrag

### ISSUE THREAD

**Issue by JakobWegmann** [iza-institute-of-labor-economics/gettsim#224]
According to [Stefan Bach](https://twitter.com/SBachTax/status/1279302743392387073), the favorability check (Kinderfreibetrag/ Kindergeld) is done for each child separately. This is not implemented in `_beantrage_kind_freib_tu`.

**Comment by Eric-Sommer** [comment]
Thanks for the note. Certainly something to keep in mind when modelling children outside the household.

**Comment by JakobWegmann** [comment]
As I understand it, the separate favorability check is done for all children (not only "Zählkinder").

**Comment by Eric-Sommer** [comment]
In which case would the current procedure be wrong in your view? We compare child benefits with child tax allowance. Whether we do that individually or on the tax unit as a whole should not make a difference when all eligible children are in the household. We obviously account for different child benefit amounts for the 3rd and 4th child.

**Comment by JakobWegmann** [comment]
What I had in mind:
Comparing a household with two children to a household with one child, for the first child the tipping point when child tax allowance is more favorable than child benefits is at the same income: At this income, the tax advantage of one child tax allowance is equal to the child benefit of one child. However, due to the progressivity of the tax system, the tax advantage of the child tax allowance for the second child is smaller. Therefore, the tipping point for the second child is at a higher income.
Accordingly, households may profit from child tax allowance and child benefits at the same time.

**Comment by mjbloemer** [comment]
My intuition would be that there must be an order.

https://www.smartsteuer.de/online/lexikon/f/familienleistungsausgleich/#D063032800004 says:
> Bei einer Mehrzahl von Kindern ist eine isolierte Günstigerrechnung für jedes Kind, beginnend mit dem ältesten Kind, durchzuführen (R 31 Abs. 1 Satz 3 EStR). 

Isn't the word "isoliert" a bit misleading as the favorability check of one child will actually depend on the favorability outcome for the other childen? The citation says by age. However by which variable the ordering is done might be irrelevant for the total amount, as long as it is done serially.

**Comment by JakobWegmann** [comment]
> Isn't the word "isoliert" a bit misleading as the favorability check of one child will actually depend on the favorability outcome for the other childen?

"Individuell" would fit better.

> However by which variable the ordering is done might be irrelevant for the total amount, as long as it is done serially.

I think so. What happens with twins? :D

**Comment by mjbloemer** [comment]
> What happens with twins? :D

age of twins differ ;)

**Comment by hmgaudecker** [comment]
Was this addressed in #463, by any chance?

**Comment by ChristianZimpelmann** [comment]
> Was this addressed in https://github.com/iza-institute-of-labor-economics/gettsim/pull/463, by any chance?

No. But should be easier to implement no. Not the highest priority though, I guess.

**Comment by hmgaudecker** [comment]
@MImmesberger, another check whether we are doing this correctly already?

**Comment by MImmesberger** [comment]
No, we still check on `_sn` level.
